### PR TITLE
feat: publish with metadata

### PR DESCRIPTION
**Pull request by fmaule**
Using an (apparently undocumented) PUT with `multipart-form` ([source](https://community.cloudflare.com/t/how-do-you-use-the-api-to-bind-an-instance-of-a-kv-namespace-to-access-its-data-in-a-worker/361280/4)), add metadata (secrets and KV bindings) to the deployment.

Still a WIP (need to parse `.toml` file?)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,26 +1,95 @@
...
+    throw new Error('Missing parameter');
+  }
+
+  const tomlFile = await fs.readFile(wranglerTomlPath);
+  const tomlData = toml.parse(tomlFile);
+
+  const { env: tomlEnvs } = tomlData;
+  const envConfig = tomlEnvs[environment];
+
+  // TODO: can this be an array in the TOML file?
```
Comment (ksnll) [inline src/index.js:48]: it looks like it's just an object from the Cloudflare docs. I would remove this comment

File: src/index.js [hunk 2]
```diff
@@ -1,26 +1,95 @@
...
+  const envVars = Object.entries(envConfig.vars).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
+      ...{ name: key, text: value, type: 'plain_text' },
```
Comment (ksnll) [inline src/index.js:52]: don't know much about the data format, but it looks sketchy. Isn't this overriding name text and type every cycle?

File: src/index.js [hunk 3]
```diff
@@ -1,26 +1,95 @@
...
+  }));
+
+  const workerName = envConfig.name;
+  const bindings = [...[envVars], ...kvNamespaces];
+
+  core.debug(`Configuring ${[envVars].length} environment variables`);
+  core.debug([envVars].filter((env) => env.type === 'plain_text'));
+
+  core.debug(`Configuring ${kvNamespaces.length} KV namespaces`);
+  core.debug(kvNamespaces);
```
Comment (ksnll) [inline src/index.js:70]: will it work with 0 namespaces and 0 vars? what if I specify a non-existing environment?

**Review by ksnll — APPROVED** [review]
🚀 Just added some questions

File: src/index.js [hunk 4]
```diff
@@ -1,26 +1,95 @@
...
+  );
+
+  const kvNamespaces = envConfig.kv_namespaces.map((namespace) => ({
+    name: namespace.binding,
+    namespace_id: namespace.id,
+    type: 'kv_namespace',
+  }));
+
+  const workerName = envConfig.name;
+  const bindings = [...[envVars], ...kvNamespaces];
```
Comment (ksnll) [inline src/index.js:64]: is this to shallow clone? couldn't you just [envVars, ... kvNamespaces]?

File: src/index.js (hunk 2 above)
Comment (fmaule) [reply src/index.js:52]: yeah, my bad - ruins of previous stuff :D

File: src/index.js (hunk 3 above)
Comment (fmaule) [reply src/index.js:73]: good one, fixed

**Pull request merged**